tests/modules/specs/_create_modulemd.py

```diff
@@ -64,27 +64,24 @@ def parse_module_id(module_id):
             rpms_with_epoch.append(nevra)
         rpms = rpms_with_epoch
 
-        mmd = Modulemd.Module()
-        mmd.set_mdversion(int(1))
-        mmd.set_name(name)
-        mmd.set_stream(stream)
-        mmd.set_version(int(version))
-        sset = Modulemd.SimpleSet()
-        sset.add("LGPLv2")
-        mmd.set_module_licenses(sset)
-        mmd.set_summary("Fake module")
-        mmd.set_description(mmd.peek_summary())
-        artifacts = Modulemd.SimpleSet()
+        module_stream = Modulemd.ModuleStreamV2.new(name, stream)
+        module_stream.set_version(int(version))
+        module_stream.add_module_license("LGPLv2")
+        module_stream.set_summary("Fake module")
+        module_stream.set_description(module_stream.get_summary())
         for rpm in rpms:
-            artifacts.add(rpm[:-4])
-        mmd.set_rpm_artifacts(artifacts)
+            module_stream.add_rpm_artifact(rpm[:-4])
         for profile_name in profiles:
-            profile = Modulemd.Profile()
-            profile.set_name(profile_name)
+            profile = Modulemd.Profile.new(profile_name)
             profile.set_description("Description for profile %s." % profile_name)
-            profile_rpms = Modulemd.SimpleSet()
-            profile_rpms.set(profiles[profile_name]["rpms"])
-            profile.set_rpms(profile_rpms)
-            mmd.add_profile(profile)
 
-        Modulemd.dump([mmd], os.path.join(module_dir, "%s.%s.yaml" % (module_id, arch)))
+            for profile_rpm in profiles[profile_name]["rpms"]:
+                profile.add_rpm(profile_rpm)
+
+            module_stream.add_profile(profile)
+
+        module_index = Modulemd.ModuleIndex()
+        module_index.add_module_stream(module_stream)
+
+        with open(os.path.join(module_dir, "%s.%s.yaml" % (module_id, arch)), 'w') as f:
```

Comment:
We actually might be able to simplify this nicely, if you add `modulemd` to the name of the created yaml files createrepo_c since version 0.16.0 will find them automatically and add them to the repository, so we can get rid of `_createrepo_c_modularity_hack.py ` completely.
```suggestion
 with open(os.path.join(module_dir, "%s.%s.modulemd.yaml" % (module_id, arch)), 'w') as f:
```

Reply:
Would you mind terribly if we committed this patch as-is and you (or someone on your team) handles that part?

I don't have the cycles to spare at the moment.

Reply:
Sure, that is no problem, thanks for the patch. :slightly_smiling_face: